api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,222 @@
+//
+// 2. **Zero Matching CountForLauncher**
+//   - When no CountForLauncher matches a specific (Node, AcceleratorSet, LauncherConfig) tuple:
+//     - No pre-provisioning occurs for that specific combination
+//     - Launcher pods are created on-demand when needed
+//
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=lpp
+
+type LauncherPoolPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LauncherPoolPolicySpec   `json:"spec,omitempty"`
+	Status LauncherPoolPolicyStatus `json:"status,omitempty"`
+}
+
+// LauncherPoolPolicySpec defines the node-level idle pool configuration.
+type LauncherPoolPolicySpec struct {
+	// LauncherPoolForNodeType defines pool spec per node type.
+	LauncherPoolForNodeType []LauncherPoolForNodeType `json:"launcherPoolForNodeType"`
+}
+
+// LauncherPoolForNodeType defines launcher count for a class of nodes.
+type LauncherPoolForNodeType struct {
+	// Selector describes the hardware characteristics of target nodes.
+	//
+	// Introduce an EnhancedNodeSelector that supports combining label-based
+	// matching with resource field conditions.
+	// For example:
+	// enhancedNodeSelector:
+	//  # 1. Label selector (compatible with existing metav1.LabelSelector)
+	//  labelSelector:
+	//    matchLabels:
+	//      nvidia.com/gpu.family: ada-lovelace
+	//    matchExpressions:
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      cpu:
+	//        min: "16"
+	//        max: "64"
+	//      memory:
+	//        min: "128Gi"
+	//        max: "512Gi"
+	//      accelerators:
+	//        nvidia.com/gpu:
+	//          min: "2"
+	//	        max: "8"
+	//  acceleratorSelector:
+	//    type: "nvidia.com/gpu"
+	//    memory: "48Gi"
+	//    count: 4
+	// +required
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// PerAcceleratorCount defines pre-configuration quantities for each accelerator type
+	PerAcceleratorCount []PerAcceleratorCount `json:"perAcceleratorCount"`
+}
+
+// EnhancedNodeSelector defines node selector with label selector and resource requirements.
+type EnhancedNodeSelector struct {
+	// LabelSelector defines the label selector for a node.
+	// +required
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+	// ResourceRequirements defines the resource requirements for a node.
+	// +optional
+	ResourceRequirements *ResourceRequirements `json:"resourceRequirements,omitempty"`
+	// AcceleratorSelector defines accelerator-specific selection criteria
+	AcceleratorSelector *AcceleratorSelector `json:"acceleratorSelector,omitempty"`
+}
+
+// PerAcceleratorCount defines configuration for specific accelerators
```

Comment:
Oh, I am sorry, I did not think through my previous suggestion of ordering very well. The declarations in this file form a tree. For a tree, I think that the best ordering is a pre-order traversal.